connector-pagerduty/src/main/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyCloudEventDataExtractor.java

```diff
@@ -0,0 +1,53 @@
+package com.redhat.cloud.notifications.connector.pagerduty;
+
+import com.redhat.cloud.notifications.connector.CloudEventDataExtractor;
+import com.redhat.cloud.notifications.connector.authentication.AuthenticationDataExtractor;
+import io.vertx.core.json.JsonObject;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.camel.Exchange;
+import org.apache.commons.validator.routines.UrlValidator;
+
+import java.util.MissingResourceException;
+
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.TARGET_URL;
+import static org.apache.commons.validator.routines.UrlValidator.ALLOW_LOCAL_URLS;
+
+@ApplicationScoped
+public class PagerDutyCloudEventDataExtractor extends CloudEventDataExtractor {
+
+    public static final String AUTHENTICATION = "authentication";
+    public static final String URL = "url";
+
+    // HTTP URLs (or disabling SSL verification) must be permitted to run test cases
+    private static final UrlValidator URL_VALIDATOR = new UrlValidator(new String[]{"http", "https"}, ALLOW_LOCAL_URLS);
+
+    @Inject
+    AuthenticationDataExtractor authenticationDataExtractor;
+
+    @Override
+    public void extract(Exchange exchange, JsonObject cloudEventData) throws IllegalArgumentException {
+
+        validatePayload(cloudEventData);
+
+        exchange.setProperty(TARGET_URL, cloudEventData.getString(URL));
+
+        JsonObject authentication = cloudEventData.getJsonObject(AUTHENTICATION);
+        authenticationDataExtractor.extract(exchange, authentication);
+
+        cloudEventData.remove(URL);
+        cloudEventData.remove(AUTHENTICATION);
+
+        exchange.getIn().setBody(cloudEventData.encode());
```

Comment:
Instead of doing this, we could set the `exchange` body to `cloudEventData.getJsonObject(PAYLOAD)`. We wouldn't need to remove `URL` and `AUTHENTICATION`. `PagerDutyTransformer` would also be able to consume `exchange.getIn().getBody(JsonObject.class)` as the payload.

The current code is fine for now. It can be improved in a subsequent PR.

Reply:
There's one other field in the cloud event that needs to be set, `.environment_url`. I might move it to the payload to try to fix test case issues.

Reply:
and it turns out the encoding was breaking my test cases, so thanks for the tip 😅